src/Internal/PluginManagement/TranslationsLoader.php

```diff
@@ -0,0 +1,165 @@
+	 */
+	public function init_hooks() {
+		$this->hooks_proxy->add_filter( 'pre_set_site_transient_update_plugins', [ $this, 'load_wcpay_translations' ] );
+	}
+
+	/**
+	 * Hooks into auto-update process to load plugin translations from translate.wordpress.com.
+	 *
+	 * Runs in a cron thread, or in a visitor thread if triggered
+	 * by _maybe_update_plugins(), or in an auto-update thread.
+	 *
+	 * @param object $transient The update_plugins transient object.
+	 *
+	 * @return object The same or a modified version of the transient.
+	 */
+	public function load_wcpay_translations( $transient ) {
+		try {
+			if ( is_object( $transient ) ) {
+				$translations            = $this->get_translations_update_data();
+				$merged_translations     = array_merge( isset( $transient->translations ) ? $transient->translations : [], $translations );
+				$transient->translations = $merged_translations;
+			}
+		} catch ( \Exception $ex ) {
+			$this->logger->error( 'Error with loading WooPayments translations from WordPress.com. Reason: ' . $ex->getMessage() );
+			return $transient;
+		}
+		return $transient;
+	}
+
+	/**
+	 * Get translations updates information.
+	 *
+	 * @return array Update data {product_id => data}
+	 */
+	public function get_translations_update_data() {
+		$installed_translations = wp_get_installed_translations( 'plugins' );
+		$locales                = array_values( get_available_languages() );
+
+		if ( empty( $locales ) ) {
+			return [];
+		}
+
+		// Use the same timeout values as Woo Core https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php#L257.
+		$timeout = wp_doing_cron() ? 30 : 3;
+
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		$plugin_data = get_plugin_data( WCPAY_PLUGIN_FILE );
+
+		/**
+		 * Note: TextDomain could differ from the plugin slug, but WordPress uses TextDomain to load translations.
+		 */
+		$plugin_name = $plugin_data['TextDomain'];
+
+		$request_body = [
+			'locales' => $locales,
+			'plugins' => [],
+		];
+
+		$request_body['plugins'][ $plugin_name ] = [
+			'version' => WCPAY_VERSION_NUMBER,
+		];
+
+		$raw_response = wp_remote_post(
+			'https://translate.wordpress.com/api/translations-updates/woocommerce',
+			[
+				'body'    => wp_json_encode( $request_body ),
+				'headers' => [ 'Content-Type: application/json' ],
+				'timeout' => $timeout,
+			]
+		);
+
+		// Something wrong happened on the translate server side.
+		$response_code = wp_remote_retrieve_response_code( $raw_response );
+		if ( 200 !== $response_code ) {
+			return [];
+		}
+
+		$response = json_decode( wp_remote_retrieve_body( $raw_response ), true );
+
+		// API error, api returned but something was wrong.
+		if ( array_key_exists( 'success', $response ) && false === $response['success'] ) {
```

Comment:
Shouldn't this verify the format of the response?

```suggestion
		if ( ! array_key_exists( 'success', $response ) || false === $response['success'] ) {
```

Reply:
Good point, Rado. I've added exceptions in https://github.com/Automattic/woocommerce-payments/pull/8050/commits/a770bb637c2ffc1834d95fe96323bd5ce91742bb